CMSgov/easi-app
# Easi 3048/it gov admin status be

# [EASI-3048](https://jiraent.cms.gov/browse/EASI-3048)

## Changes and Description

- Introduce `statusAdmin` on the SystemIntake  GQL type
- Introduce a resolver file to calculate the admin status that can be displayed
    - Introduce unit tests to cover the possible states
             - Introduce a table driven testing model for this resolver to handle possible combinations.



## How to test this change
1. Bring up the app and make a system intake.
```gql
mutation createSystemIntake{
  
  createSystemIntake(input:{
      requestType :MAJOR_CHANGES
      requester: {
          name: "Mr. Fun Person"
      }
    } )
  
  {
    id
 }
}
```
2. Get the system intake with this query and verify that statusAdmin returns `INITIAL_REQUEST_FORM_IN_PROGRESS`


``` GQL
query systemIntake{
  
  systemIntake(id:"{{systemIntakeID}}")
  {
    statusAdmin
  }
}
```



## Other things to look at
1. Do the unit tests make sense to you?
2. Does it make sense to you for the resolver to return a status, and an error code?

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
